application/views/ride_views.py

```diff
@@ -263,23 +262,50 @@ def put(self, rideId, requestId):
         """Driver can accept or reject the ride offer."""
         try:
             data = request.get_json()
-            act = True
+            action = False
             response = ''
+            # check whether driver already accepted the offer.
+            query = "select status from requests where req_id='{}'".format(requestId)
+            result = db.execute(query)
+            status = result.fetchone()
+
+            query = "select available_space from rides where ride_id='{}'"\
+                        .format((rideId))
+            result = db.execute(query)
+            seats = result.fetchone()
+            available_seats = seats[0]
+
+            # check for action to take
             if data['action'].lower() == 'accept':
-                act = True
-                response = {'message': 'Request accept'}
+                if status[0]:
+                    return {'message': 'You already accepted this user request'},403
+                action = True                
+                # Decrement the available seats by one
+                available_seats -= 1
+                
+                # set message to be returned to user after request update cycle
+                response = {'message': 'Request accepted'}
             else:
-                act = False
+                # Reject an already accepted request; this means available seats should be incremented
+                if status[0]:
+                    available_seats += 1
+                else:
+                    return {'message': 'Request is rejected'}
+                    
                 response = {'message': 'Request rejected'}
+
             query = "UPDATE requests SET status = '{}'\
-             where requests.req_id = '{}' and requests.ride_id = '{}'" \
-             . format(act, int(requestId), int(rideId))
-            cursor.execute(query)
+             where requests.req_id = '{}'" \
+             . format(action, (requestId))
+            db.execute(query)
+
+            query = "update rides set available_space='{}' where ride_id='{}'"\
+                        .format(int(available_seats), rideId)
+            db.execute(query)
+
```

Comment:
Are both of the above query statements working?

Reply:
Yes. They are working. I tested them using postman